### PR TITLE
Support AccessDeniedPath for Cookie 403 redirection

### PR DESCRIPTION
**Pull request by HaoK**
Fixes https://github.com/aspnet/Security/issues/166

cc @Praburaj @Tratcher @lodejard

### CODE REVIEW

File: src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs [hunk 1]
```diff
@@ -323,6 +323,41 @@ private static bool IsHostRelative(string path)
...
+                            Request.Host +
+                            Request.PathBase +
+                            Options.AccessDeniedPath;
+
+                        var redirectContext = new CookieApplyRedirectContext(Context, Options, accessDeniedUri);
+                        Options.Notifications.ApplyRedirect(redirectContext);
+                    }
+                    catch (Exception exception)
+                    {
+                        CookieExceptionContext exceptionContext = new CookieExceptionContext(Context, Options,
```
Comment (Praburaj) [inline src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs]: minor `var`.

File: test/Microsoft.AspNet.Authentication.Test/Cookies/CookieMiddlewareTests.cs [hunk 2]
```diff
@@ -421,6 +421,27 @@ private Task SignOutAsWrong(HttpContext context)
         }
 
         [Fact]
+        public async Task CookieTurns401ToAccessDeniedWhenSetAndIfAuthenticated()
+        {
+            var clock = new TestClock();
+            TestServer server = CreateServer(options =>
+            {
+                options.SystemClock = clock;
+                options.AccessDeniedPath = new PathString("/accessdenied");
```
Comment (Praburaj) [inline test/Microsoft.AspNet.Authentication.Test/Cookies/CookieMiddlewareTests.cs]: Can we also add a test for the default value of `AccessDeniedPath`. Client can check if it received a `403` status code.

**Comment by Praburaj** [comment]
Looks :shipit: good to me other than above comments.

File: test/Microsoft.AspNet.Authentication.Test/Cookies/CookieMiddlewareTests.cs (hunk 2 above)
Comment (HaoK) [inline test/Microsoft.AspNet.Authentication.Test/Cookies/CookieMiddlewareTests.cs]: That test already existed, https://github.com/aspnet/Security/blob/3-3accessdenied/test/Microsoft.AspNet.Authentication.Test/Cookies/CookieMiddlewareTests.cs#L406

Comment (Praburaj) [inline test/Microsoft.AspNet.Authentication.Test/Cookies/CookieMiddlewareTests.cs]: sounds good.

**Comment by HaoK** [comment]
Merged e2bb76280f07404f23ecbe8f4b2481082692abc2

**Pull request closed without merging**